code-challenges.js

```diff
@@ -17,7 +17,36 @@ const fruit3 = "cherry"
 const fruit4 = "kiwi"
 // Expected outcome: "cherry"
 
+// input: two strings
+// output: one string containing the most chars
+// logic:
+// create function
+// .length property to determine length of each string
+// use conditional if statement to evaluate which string is longer
+
+const longestStr = (str1, str2) => {
+    // validate and return early
+    if (typeof str1 !== "string" || typeof str2 !== "string") {
+        return;
+    }
+    // .length returns the number of total characters in strings including spaces and will start at 1
+    let str1Length = str1.length;
+    let str2Length = str2.length;
+    // conditional statement. if one string is not longer than the other, they must be equal in length
+    if (str1Length > str2Length) {
+        return str1;
+    } else if (str2Length > str1Length) {
+        return str2;
+    } else {
+        return `${str1} and ${str2} have the same amount of characters`;
+    }
+}
+console.log(longestStr(fruit1, fruit2)); // output: banana
+
 // Explain your code:
+// If two strings are inputed, this function will return the longer of the two strings. If any other data type is inputed, this function will return immediately.
+
+// After having just completed this example I've realized we weren't meant to do it again (sorry if you said this) 🙃
```

Comment:
I love how you checked for datatype!  It's a great practice. You can also have your else statement return some sort of error message and it will handle the wrong data types. :) Excellent work!